Row.c

```diff
@@ -454,6 +454,13 @@ void Row_printNanoseconds(RichString* str, unsigned long long totalNanoseconds,
    Row_printTime(str, totalHundredths, coloring);
 }
 
+static inline unsigned char Row_getPrecision (double displayRate){
+   if (displayRate < 10) return 3;
+   else if (displayRate < 100) return 2;
+   else if (displayRate < 1000) return 1;
```

Comment:
Consider there is a rounding issue with this approach. Look at `Meter_humanUnit()` function to see how I addressed that.